src/pint/models/timing_model.py

```diff
@@ -2534,6 +2534,32 @@ def validate_toas(self, toas):
         if bad_parameters:
             raise MissingTOAs(bad_parameters)
 
+    def find_empty_masks(self, toas, freeze=False):
+        """Find unfrozen mask parameters with no TOAs before trying to fit
+
+        Parameters
+        ----------
+        toas : pint.toa.TOAs
+        freeze : bool, optional
+            Should the parameters with on TOAs be frozen
+
+        Returns
+        -------
+        list
+            Parameters with no TOAs
+        """
+        bad_parameters = []
+        for maskpar in self.get_params_of_type_top("maskParameter"):
+            par = getattr(self, maskpar)
+            if "TNEQ" in str(par.name) or par.frozen:
+                continue
+            if len(par.select_toa_mask(toas)) == 0:
+                bad_parameters.append(maskpar)
+                if freeze:
+                    log.info(f"'{maskpar}' has no TOAs so freezing")
```

Comment:
Since this is explicitly changing something that the user requested (accidentally or not), I think this should be a warning and not an info.

Reply:
Except the default is to not change.  Change only happens when the user sets `freeze=True`

Reply:
Ah.  I missed that.  So this isn't called by default anywhere?  The user has to explicitly call this before a fit?  If so, then I agree with your plan (and apologize!).